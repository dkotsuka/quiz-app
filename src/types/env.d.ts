declare module '@env' {
    export const API_URL: string;
    export const API_TOKEN_URL: string;
    export const API_CATEGORY_URL: string;
}