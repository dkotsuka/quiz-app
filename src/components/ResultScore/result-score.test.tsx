import React from 'react'
import { ResultScore } from '.'
import { create } from 'react-test-renderer';
import { Score } from '../../screens/QuestionScreen/question-screen.types';

describe("ResultScore", () => {
    const score: Score = {
        easy: [1,2],
        medium: [3,4],
        hard: [5,6]
    }
    
    const resultScore = create(<ResultScore score={score} />)

    it("maches snapshot", () => {
        expect(resultScore).toMatchSnapshot()
    })

    it("shows correctly", () => {
        const correctText = resultScore.root.findByProps({testID:"result-score-successes-text"}).props
        const wrongText = resultScore.root.findByProps({testID:"result-score-mistakes-text"}).props

        expect(correctText.children).toEqual(9)
        expect(wrongText.children).toEqual(12)
    })
})