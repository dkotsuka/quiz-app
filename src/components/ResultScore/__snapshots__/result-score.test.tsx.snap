// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultScore maches snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "flex-start",
        "backgroundColor": "#EFEFF2",
        "borderRadius": 10,
        "flexDirection": "row",
        "justifyContent": "space-around",
        "marginHorizontal": 60,
        "paddingBottom": 10,
        "paddingLeft": 10,
        "paddingRight": 10,
        "paddingTop": 10,
      },
    ]
  }
>
  <View
    alignItems="center"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto Bold",
            "fontSize": 20,
          },
          Object {
            "fontSize": 28,
          },
        ]
      }
      testID="result-score-successes-text"
    >
      9
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto",
            "fontSize": 14,
          },
          Object {
            "fontSize": 13,
          },
        ]
      }
    >
      Successes
    </Text>
  </View>
  <View
    alignItems="center"
    style={
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "column",
          "justifyContent": "flex-start",
        },
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto Bold",
            "fontSize": 20,
          },
          Object {
            "fontSize": 28,
          },
        ]
      }
      testID="result-score-mistakes-text"
    >
      12
    </Text>
    <Text
      style={
        Array [
          Object {
            "fontFamily": "Roboto",
            "fontSize": 14,
          },
          Object {
            "fontSize": 13,
          },
        ]
      }
    >
      Mistakes
    </Text>
  </View>
</View>
`;
